comp/core/workloadmeta/types.go

```diff
@@ -446,10 +454,60 @@ func (o OrchestratorContainer) String(_ bool) string {
 	return fmt.Sprintln("Name:", o.Name, "ID:", o.ID)
 }
 
+// ECSContainer is a reference to a container running in ECS
+type ECSContainer struct {
+	DisplayName   string
+	Networks      []ContainerNetwork
+	Volumes       []ContainerVolume
+	Health        *ContainerHealthStatus
+	DesiredStatus string
+	KnownStatus   string
+	Type          string
+	LogDriver     string
+	LogOptions    map[string]string
+	ContainerARN  string
+	Snapshotter   string
+}
+
+// String returns a string representation of ECSContainer.
+func (e ECSContainer) String(verbose bool) string {
+	var sb strings.Builder
+
+	if len(e.Volumes) > 0 {
+		_, _ = fmt.Fprintln(&sb, "----------- Volumes -----------")
+		for _, v := range e.Volumes {
+			_, _ = fmt.Fprint(&sb, v.String(verbose))
+		}
+	}
+
+	if len(e.Networks) > 0 {
+		_, _ = fmt.Fprintln(&sb, "----------- Networks -----------")
+		for _, n := range e.Networks {
+			_, _ = fmt.Fprint(&sb, n.String(verbose))
+		}
+	}
+
+	if e.Health != nil {
+		_, _ = fmt.Fprintln(&sb, "----------- ECS Container Health -----------")
+		_, _ = fmt.Fprint(&sb, e.Health.String(verbose))
+	}
+	_, _ = fmt.Fprintln(&sb, "----------- ECS Container Priorities -----------")
+	_, _ = fmt.Fprintln(&sb, "DesiredStatus:", e.DesiredStatus)
+	_, _ = fmt.Fprintln(&sb, "KnownStatus:", e.KnownStatus)
+	_, _ = fmt.Fprintln(&sb, "Type:", e.Type)
+	_, _ = fmt.Fprintln(&sb, "LogDriver:", e.LogDriver)
+	_, _ = fmt.Fprintln(&sb, "LogOptions:", e.LogOptions)
+	_, _ = fmt.Fprintln(&sb, "Snapshotter:", e.Snapshotter)
+
+	return sb.String()
+}
+
 // Container is an Entity representing a containerized workload.
 type Container struct {
 	EntityID
 	EntityMeta
+	// ECSContainer contains properties specific to container running in ECS
+	*ECSContainer
```

Comment:
As we are embedding `ECSContainer` as a pointer, I think you should modify the `Merge` method of the `Container` entity to avoid variable aliasing.